src/mesh.cc

```diff
@@ -1,96 +1,84 @@
-
 #include "mesh.h"
 
+bool pipenetwork::Mesh::create_junctions(
+    const std::vector<Index>& ids, const std::vector<double>& elevations,
+    const std::vector<double>& demands,
+    const std::vector<double>& leak_diameters) {
 
-// Create nodal pointers and assign indices based on input coordinates
-void pipenetwork::Mesh::create_nodes(
-    const std::vector<Eigen::Vector3d>& coords) {
-  Index index = 0;
-  for (const auto& coord : coords) {
-    nodes_.emplace(std::pair<Index, std::shared_ptr<pipenetwork::Node>>(
-        index, std::make_shared<pipenetwork::Node>(index, coord)));
-    ++index;
+  bool stat{false};
+  Index i = 0;
+  for (const auto& id : ids) {
+    nodes_.emplace(id, std::make_shared<pipenetwork::Junction>(
+                           id, elevations[i], demands[i], leak_diameters[i]));
+    ++i;
   }
+  return stat;
+}
+
+bool pipenetwork::Mesh::create_reservoirs(const std::vector<Index>& ids,
+                                          const std::vector<double>& heads) {
+  Index i = 0;
+  bool stat{false};
+  for (const auto& id : ids) {
+    nodes_.emplace(id, std::make_shared<pipenetwork::Reservoir>(id, heads[i]));
+    ++i;
+  }
+  stat = true;
```

Comment:
Do you expect it to fail anywhere here? why bool?

Reply:
Yeah its not necessary. I will make it void